portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,73 @@
+const api = ""
+
+window.addEventListener("DOMContentLoaded", () => {
+
+    setMessages();
+
+    const send_button = document.getElementById('send-button');
+    const text_input = document.getElementById('send-input');
+
+
+
+    send_button.addEventListener('click', () => {
+        postMessage(text_input.value);
+        text_input.value = "";
+    });
+    text_input.addEventListener('keyup', (event) => {
+        if (event.keyCode === 13) {
+            postMessage(text_input.value);
+            text_input.value = "";
+        }
+    })
+
+
+
+});
+
+
+
+
+
+
+function appendMessage({ text }) {
+    const container = document.getElementById('messages');
+    const message_div = document.createElement('div')
+    message_div.classList.add('card');
+    message_div.classList.add('msg-bubble');
+    message_div.innerText = text;
+    container.appendChild(message_div);
+}
+
+
+async function setMessages() {
+    const messages = await getMessages();
+    console.log('the messages ', messages);
+    messages.forEach(appendMessage);
+}
+
+async function getMessages() {
+    const res = await fetch(`${api}/comments`, {
+        method: 'GET'
+    });
+
+    return await res.json();
+}
+
+
+async function postMessage(text) {
+
+    try {
+
+        const params = new URLSearchParams();
+        params.append('text', text);
+        appendMessage({ text });
```

Comment:
This works but has some drawbacks:

 - You're appending the message before it has been sent to the server/stored in the DataStore
 - You're not making all fields available to the 'appendMessage' function, so if you wanted to also print e.g. the timestamp (or an "edit" modal using the comment `id`) it won't work for these new comments since only `text` is set.

This would be more predictable if you use the server response (`res`) and append that instead.